File: ruby/daily-top10.rb
#!/usr/bin/ruby

require 'csv'

Dir.open('.').grep(/.*\.csv$/).sort.each do |filename|
  counter = Hash.new(0)
  CSV.foreach(filename) do |row|
    counter[row[6]] += 1
  end
  puts filename
  counter.sort_by{|k, v| -v}.slice(1..10).each do |name,value|
    puts "#{name},#{value}"
  end
end

File: perl/daily-top10.pl
#!/usr/bin/perl

use strict;

opendir(my $DIR,".");
foreach my $filename ( sort( grep(/.*\.csv$/, readdir($DIR) ) ) ){
    my %counter;
    open(my $FILE,$filename);
    while(my $line = <$FILE>) {
	my $name = (split(/,/, $line))[6];
	++$counter{$name};
    }
    close($FILE);
    print "$filename\n";
    my @names = sort {$counter{$b} <=> $counter{$a}} keys(%counter);
    foreach my $name (splice(@names, 0, 10)) {
	print "$name,$counter{$name}\n";
    }
}
closedir($DIR);
exit(0);
